internal/ldcache/ldcache.go

```diff
@@ -105,14 +101,7 @@ func New(logger logger.Interface, root string) (LDCache, error) {
 
 	logger.Debugf("Opening ld.conf at %v", path)
 	f, err := os.Open(path)
-	if os.IsNotExist(err) {
-		logger.Warningf("Could not find ld.so.cache at %v; creating empty cache", path)
-		e := &empty{
-			logger: logger,
-			path:   path,
-		}
-		return e, nil
-	} else if err != nil {
+	if err != nil {
```

Comment:
So this change implies that we error out unconditionally if there is not an ld.s.cache file now?

Reply:
Yes, but this is handled in the caller now:
```
		b.logger.Warningf("Failed to load ldcache: %v", err)
		if b.isOptional {
			return &null{}
		}
		return &notFound{}
```

Reply:
The original implementation was mixing some of the concepts from finding libraries with just reading the ldcache. This has now been simplified so that this package only LOADS the cache and returns its contents as a list. The non-existence etc. is now handled higher up the call stack.

Reply:
Ack